src/odemis/acq/stream/_sync.py

```diff
@@ -1088,20 +1137,64 @@ def _runAcquisitionEbeam(self, future):
             self._current_future = None
             self._acq_done.set()
 
-    def _waitForImage(self, px_time):
+    def _integrateImages(self, n):
+        """
+        Integrate/sum the last n acquired images in self._acq_data and replace by the
+        integrated image. If values overflow, clip the image.
+        :param n: (int) Number of images that need to be integrated (summed).
+        """
+        # TODO is there an sem_idx or similar? How to handle if more than 1 sem det -> for loop?
```

Comment:
Should the fast path be here or where the `integrateImages` is called from? I think in the polarimetry pr, you suggested to put it in the caller?

Should the `MD_INTEGRATION_COUNT` also be appended when =1?

Is there always only one SEM stream and one detector stream? 
Is there a `sem_idx` similar to `self._ccd_idx`

Reply:
Whether the `if` is inside the function or in the caller, of course, eventually it's the same behaviour. So it's a little bit a matter of taste and I think also a matter of the "semantic". Personally, I'd say that here it's just a "fast path" (because the standard path would still work fine with n == 1), so it'd feel more logical to hide this implementation detail inside the function. In the case of the polarimetry export, it'd read more like `draw_logo(None)`, where the code can simply not work with that special parameter value.

The default value of MD_INTEGRATION_COUNT is 1. So I don't think you need to put it explicitly. I'd suggest to only put it if that simplifies the code (which I don't think is the case).

There is no `sem_idx`, because the idea is that every stream which is not `_ccd_idx` should be considered a SEM stream. BTW, in that function, I think that if you use MD_DET_TYPE, you don't even need to care about the SEM vs CCD.